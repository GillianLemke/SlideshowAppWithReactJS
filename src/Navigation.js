import React, { Component } from 'react';
import './App.css';
import $ from 'jquery';
//import Presentation from './Presentation';

export default class Navigation extends Component {
  constructor() {
    super();
    this.state = {
      title: 'presentation title',
      slideTitles: ['slide1', 'slide2', 'slide3'],
      slideVisited: [true, false, false],
    };
  }

  findListed() {
    // how to return each item in the list
    for(var i = 0; i < this.state.slideTitles.length; i++) {
      if (this.state.slideVisited[i]) {
        return (<p>&#9989;&nbsp;{this.state.slideTitles[i]}</p>);
      } else {
        return (<p>{this.state.slideTitles[i]}</p>);
      }
    }
  }

  getPageInfo() {
    return $.getJSON('google.com/slide')
      .then((data) => {
        // how can we get data for individual variables?
        // use regex to separate string into different variables?
        // assuming format of json is type,title,content
        // separated by an uncommon utf char
        // '^/w*' matches type (should skip over this)
        this.setState({ title:data.results })
      });
  }

  render() {
    return(
      <div>
        <h3>presentation title</h3>
        <ul>

          {this.findListed()}
        </ul>
      </div>
    );
  }
}
