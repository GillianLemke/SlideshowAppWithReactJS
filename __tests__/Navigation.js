import React from 'react'
import Navigation from '../src/Navigation.js'
import {shallow} from 'enzyme' // could also mount
import toJson from 'enzyme-to-json'

describe('Navigation', () => {
  it('renders', () => {
    const wrapper = shallow(<Navigation />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
});
