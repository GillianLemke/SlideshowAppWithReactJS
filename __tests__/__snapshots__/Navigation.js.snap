exports[`Navigation renders 1`] = `
<div>
  <h3>
    presentation title
  </h3>
  <ul>
    <p>
      ✅ 
      slide1
    </p>
  </ul>
</div>
`;
